download_test_vectors.sh

```diff
@@ -75,12 +75,18 @@ done
 
 # delete tarballs and unpacked data from old versions
 for tpath in tarballs/*; do
+	if [[ "$tpath" =~ tarballs/slashing-* ]]; then
```

Comment:
While it's probably not that important -- this is a fairly static set of tests being downloaded, and the failure modes are at least deterministic should they ever occur -- this is still essentially an O(n^2) approach to this problem, blacklisting all specifically recognized non-consensus spec test vector tarballs rather than recognize only consensus spec test vector tarballs. If nothing else, it somewhat mixes conceptual concerns insofar as in theory, this script shouldn't care about the slashing test vectors either way.

Because the consensus spec test vectors are named ... uninformatively, for this purpose (`v1.3.0-alpha.1` for example, with no description of what it's a version of), there's some utility in this approach, but another possible balance is to modify `for tpath in tarballs/*` and `tpath in tests-*` a few lines later to instead iterate over only some patterns which basically match the consensus spec test vectors (e..g., `for tpath in tarballs/*`).

One can be more precise, but at the cost of probably unwarranted-for-now-complexity, by building a pattern from all supported versions. Otherwise, doing so creates a lingering issue/coupling over which to trip if the version numbering shifts enough.

Also, as much as feasible (sometimes, it's not as much, I've been trying to "quote arguments" as a kind of `shellcheck` linting-friendliness, even if one can look at the actual values which have thus far occurred which might fill those variables and ascertain as a human they won't contain spaces.

Reply:
It's not doing an `ls` here, just a wildcard string compare (previously, regex but that one only worked by accident)

```bash
if [[ "asdfasdf" == "asdfa"* ]]; then echo "yes"; else echo "no"; fi
```

Reply:
and yeah, another fix would be to rename consensus tarballs so that the `for tpath in tarballs/*` doesn't list the slashing tests as well. wanted to focus on the fix for now, over introducing rename with potential side effects

Reply:
Reasonable enough.